exercises/05.actions/05.solution.history-revalidation/server/app.js

```diff
@@ -70,23 +76,27 @@ app.post('/action/:shipId?', bodyParser.text(), async (req, res) => {
 		throw new Error('Invalid action')
 	}
 
-	const bb = busboy({ headers: req.headers })
-	const reply = decodeReplyFromBusboy(bb, moduleBasePath)
-	req.pipe(bb)
-	const args = await reply
+	const formData = await c.req.formData()
+	const args = decodeReply(formData, moduleBasePath)
 	const result = await action(...args)
+	return await renderApp(res, result)
+})
 
-	await renderApp(res, result)
+app.get(['/', '/:shipId?'], async c => {
+	const html = await readFile('./public/index.html', 'utf8')
+	return c.html(html, 200)
 })
 
-app.get('/:shipId?', async (req, res) => {
-	res.set('Content-type', 'text/html')
-	return res.sendFile('index.html', { root: 'public' })
+app.onError((err, c) => {
+	console.error('error', err)
+	return c.json({ error: true, message: 'Something went wrong' }, 500)
 })
 
-const server = app.listen(PORT, () => {
-	console.log(`🚀  We have liftoff!`)
-	console.log(`http://localhost:${PORT}`)
+console.log(`🚀  starting server at http://localhost:${PORT}`)
+const server = serve({
+	fetch: app.fetch,
+	port: PORT,
+	overrideGlobalObjects: true,
```

Comment:
Is this necessary?

Reply:
This polyfills `Request` and `Response` on Node. If you know your users will be on `Node >21` then you could set this to false and it would use the builtin values.

Reply:
NOTE: I removed the line but `true` is the default value so the "polyfilling" is still happening. Again, you can set to `false` if you are on Node >21